vpr/src/base/read_place.cpp

```diff
@@ -15,25 +16,78 @@
 #include "read_place.h"
 #include "read_xml_arch_file.h"
 
-void read_place(const char* net_file,
-                const char* place_file,
-                bool verify_file_digests,
-                const DeviceGrid& grid) {
+void read_place_header(
+    std::ifstream& placement_file,
+    const char* net_file,
+    const char* place_file,
+    bool verify_file_hashes,
+    const DeviceGrid& grid);
+
+void read_place_body(
+    std::ifstream& placement_file,
+    const char* place_file,
+    bool is_place_file);
+
+void read_place(
+    const char* net_file,
+    const char* place_file,
+    bool verify_file_digests,
+    const DeviceGrid& grid,
+    bool is_place_file) {
     std::ifstream fstream(place_file);
     if (!fstream) {
         VPR_FATAL_ERROR(VPR_ERROR_PLACE_F,
                         "'%s' - Cannot open place file.\n",
                         place_file);
     }
 
+    VTR_LOG("Reading %s.\n", place_file);
+    VTR_LOG("\n");
+
+    read_place_header(fstream, net_file, place_file, verify_file_digests, grid);
+    read_place_body(fstream, place_file, is_place_file);
+
+    VTR_LOG("Successfully read %s.\n", place_file);
+    VTR_LOG("\n");
+}
+
+void read_constraints(const char* constraints_file,
+                      bool is_place_file) {
+    std::ifstream fstream(constraints_file);
+    if (!fstream) {
+        VPR_FATAL_ERROR(VPR_ERROR_PLACE_F,
+                        "'%s' - Cannot open constraints file.\n",
+                        constraints_file);
+    }
+
+    VTR_LOG("Reading %s.\n", constraints_file);
+    VTR_LOG("\n");
+
+    read_place_body(fstream, constraints_file, is_place_file);
+
+    VTR_LOG("Successfully read %s.\n", constraints_file);
+    VTR_LOG("\n");
+}
+
+/**
+ * This function reads the header (first two lines) of a placement file.
+ * It checks whether the packed netlist file that generated the placement matches the current netlist file.
```

Comment:
Say what the header is - two lines in any order that give netlist and grid files used to generate placement. Also mention grid gives an error if it doesn't match